Fails to compile with Standards conformance mode `/permissive-`

The lastest version of the source fails to compile with Standards conformance mode [/permissive-](https://learn.microsoft.com/en-us/cpp/build/reference/permissive-standards-conformance?view=msvc-170). (Which it should if you want to make sure the code compiles properly cross compiler) 

![image](https://github.com/user-attachments/assets/cdb7210f-c147-49e4-ae1d-56827c678e15)

When I attempt to compile with the flag enabled I get the following error message:

```
FastNoise2-master\include\FastNoise\Generators\Modifiers.inl(30,31): error C2039: '__this': is not a member of 'FS_T<FastNoise::DomainOffset,FastSIMD::SSE_T<FastSIMD::Level_SSE2>>'
1>(compiling source file '../../../../src/FastSIMD/FastSIMD_Level_SSE2.cpp')
1>    \FastNoise2-master\include\FastNoise\Generators\Modifiers.inl(19,7):
1>    see declaration of 'FS_T<FastNoise::DomainOffset,FastSIMD::SSE_T<FastSIMD::Level_SSE2>>'
1>    \FastNoise2-master\include\FastNoise\Generators\Modifiers.inl(30,31):
1>    the template instantiation context (the oldest one first) is
1>        \FastNoise2-master\include\FastNoise\FastNoise_BuildList.inl(107,1):
1>        see reference to function template instantiation 'FastNoise::DomainOffset *FastSIMD::ClassFactory<FastNoise::DomainOffset,FastSIMD::Level_SSE2>(FastSIMD::MemoryAllocator)' being compiled
1>        \FastNoise2-master\src\FastSIMD\Internal\SourceBuilder.inl(12,29):
1>        see reference to variable template 'const bool is_base_of_v<FastNoise::DomainOffset,FS_T<FastNoise::DomainOffset,FastSIMD::SSE_T<4> > >' being compiled
1>        \FastNoise2-master\src\FastSIMD\Internal\SourceBuilder.inl(12,29):
1>        see reference to class template instantiation 'FS_T<FastNoise::DomainOffset,FastSIMD::SSE_T<FastSIMD::Level_SSE2>>' being compiled
1>        \FastNoise2-master\include\FastNoise\Generators\Modifiers.inl(22,5):
1>        while compiling class template member function 'FastSIMD::SSE_f32x4 FS_T<FastNoise::DomainOffset,FastSIMD::SSE_T<FastSIMD::Level_SSE2>>::Gen(FastSIMD::SSE_i32x4<FastSIMD::Level_SSE2>,FastSIMD::SSE_f32x4,FastSIMD::SSE_f32x4,FastSIMD::SSE_f32x4,FastSIMD::SSE_f32x4) const'
1>        \FastNoise2-master\include\FastNoise\Generators\Modifiers.inl(22,5):
1>        see reference to function template instantiation 'FastSIMD::SSE_f32x4 FS_T<FastNoise::DomainOffset,FastSIMD::SSE_T<FastSIMD::Level_SSE2>>::GenT<FastSIMD::SSE_f32x4,FastSIMD::SSE_f32x4,FastSIMD::SSE_f32x4,FastSIMD::SSE_f32x4>(FastSIMD::SSE_i32x4<FastSIMD::Level_SSE2>,FastSIMD::SSE_f32x4,FastSIMD::SSE_f32x4,FastSIMD::SSE_f32x4,FastSIMD::SSE_f32x4) const' being compiled
```

--- a/include/FastNoise/Generators/Modifiers.inl
+++ b/include/FastNoise/Generators/Modifiers.inl
@@ -24,12 +24,12 @@ class FS_T<FastNoise::DomainOffset, FS> : public virtual FastNoise::DomainOffset
     template<typename... P> 
     FS_INLINE float32v GenT( int32v seed, P... pos ) const
     {
-        return [this, seed]( std::remove_reference_t<P>... sourcePos, std::remove_reference_t<P>... offset )
+        return [self = this, seed]( std::remove_reference_t<P>... sourcePos, std::remove_reference_t<P>... offset )
         {
             size_t idx = 0;
-            ((offset += this->GetSourceValue( mOffset[idx++], seed, sourcePos... )), ...);
+            ((offset += self->GetSourceValue( self->mOffset[idx++], seed, sourcePos... )), ...);
 
-            return this->GetSourceValue( mSource, seed, offset... );
+            return self->GetSourceValue( self->mSource, seed, offset... );
         } (pos..., pos...);
     }
 };

